packages/Interface/src/DTK_C_API.cpp

```diff
@@ -349,98 +363,135 @@ bool DTK_is_initialized() { return DataTransferKit::isInitialized(); }
 
 void DTK_finalize() { DataTransferKit::finalize(); }
 
-void DTK_set_function( DTK_UserApplicationHandle handle, DTK_FunctionType type,
-                       void ( *f )(), void *user_data )
+int DTK_set_function( DTK_UserApplicationHandle handle, DTK_FunctionType type,
+                      void ( *f )(), void *user_data )
```

Comment:
according to [this website](http://www.cplusplus.com/reference/cerrno/errno/) best practice may be to set errno to 0 at the start of every function which may change it